src/impl.ml

```diff
@@ -577,9 +577,11 @@ let endpoint_of_string = function
       return (File_descr (fd |> int_of_string |> file_descr_of_int |> Lwt_unix.of_unix_file_descr))
     | Some "tcp", _ ->
       let host = match Uri.host uri' with None -> failwith "Please supply a host in the URI" | Some host -> host in
+      let host = Scanf.ksscanf host (fun _ _ -> host) "[%s@]" (fun elem -> elem) in
```

Comment:
Can this fail?

Reply:
from the doc of Scanf:
> as a consequence, scanning a %s conversion never raises exception End_of_file: if the end of input is reached the conversion succeeds and simply returns the characters read so far, or "" if none were ever read.

Reply:
So from my understanding if there's brackets around the ip, they'll be removed and if there's not then the string is unchanged.

Reply:
If the match fails, the function `(fun _ _ -> host)` is used to construct the return value instead.